tests/fixture/testnet/local/network.go

```diff
@@ -0,0 +1,644 @@
+			nodes = append(nodes, NewLocalNode(""))
+		}
+		ln.Nodes = nodes
+	}
+
+	// Ensure each node has keys and an associated node ID. This
+	// ensures the availability of validator node IDs for genesis
+	// generation.
+	for _, node := range ln.Nodes {
+		if err := node.EnsureKeys(); err != nil {
+			return err
+		}
+	}
+
+	// Assume all initial nodes are validator ids
+	validatorIDs := []ids.NodeID{}
+	for _, node := range ln.Nodes {
+		validatorIDs = append(validatorIDs, node.NodeID)
+	}
+
+	if keyCount > 0 {
+		// Ensure there are keys for genesis generation to fund
+		factory := secp256k1.Factory{}
+		keys := []*secp256k1.PrivateKey{}
+		for i := 0; i < keyCount; i++ {
+			key, err := factory.NewPrivateKey()
+			if err != nil {
+				return fmt.Errorf("failed to generate private key: %w", err)
+			}
+			keys = append(keys, key)
+		}
+		ln.FundedKeys = keys
+	}
+
+	if err := ln.EnsureGenesis(networkID, validatorIDs); err != nil {
+		return err
+	}
+
+	if ln.CChainConfig == nil {
+		ln.CChainConfig = LocalCChainConfig()
+	}
+
+	// Default flags need to be set in advance of node config
+	// population to ensure correct node configuration.
+	if ln.DefaultFlags == nil {
+		ln.DefaultFlags = LocalFlags()
+	}
+
+	for _, node := range ln.Nodes {
+		// Ensure the node is configured for use with the network and
+		// knows where to write its configuration.
+		if err := ln.PopulateNodeConfig(node); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Ensure the provided node has the configuration it needs to
+// start. Requires that the network has valid genesis data.
+func (ln *LocalNetwork) PopulateNodeConfig(node *LocalNode) error {
+	flags := node.Flags
+
+	// Set values common to all nodes
+	flags.SetDefaults(ln.DefaultFlags)
+	flags.SetDefaults(testnet.FlagsMap{
+		cfg.GenesisFileKey:    ln.GetGenesisPath(),
+		cfg.ChainConfigDirKey: ln.GetChainConfigDir(),
+	})
+
+	// Convert the network id to a string to ensure consistency in JSON round-tripping.
+	flags[cfg.NetworkNameKey] = fmt.Sprintf("%d", ln.Genesis.NetworkID)
+
+	// Ensure keys are added if necessary
+	if err := node.EnsureKeys(); err != nil {
+		return err
+	}
+
+	// Ensure the node's data dir is configured
+	dataDir := node.GetDataDir()
+	if len(dataDir) == 0 {
+		// NodeID will have been set by EnsureKeys
+		dataDir = filepath.Join(ln.Dir, node.NodeID.String())
+		flags[cfg.DataDirKey] = dataDir
+	}
+
+	return nil
+}
+
+// Starts a network for the first time
+func (ln *LocalNetwork) Start(w io.Writer) error {
+	if len(ln.Dir) == 0 {
+		return errLocalNetworkDirNotSet
+	}
+
+	// Ensure configuration on disk is current
+	if err := ln.WriteAll(); err != nil {
+		return err
+	}
+
+	// Accumulate bootstrap nodes such that each subsequently started
+	// node bootstraps from the nodes previously started.
+	//
+	// e.g.
+	// 1st node: no bootstrap nodes
+	// 2nd node: 1st node
+	// 3rd node: 1st and 2nd nodes
+	// ...
+	//
+	bootstrapIDs := []string{}
+	bootstrapIPs := []string{}
```

Comment:
Not sure we really care to pre-alloc stuff like this in the testing harness... But we can also pre-alloc here.